bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Region.java

```diff
@@ -660,20 +671,26 @@ public String toString () {
 	return "Region {" + zoomToHandle.entrySet().stream().map(entry -> entry.getValue() + "(zoom:" + entry.getKey() + ")").collect(Collectors.joining(","));
 }
 
+private record RegionHandle(long handle, int zoom) {
+	void destroy() {
+		OS.DeleteObject(handle);
+	}
+}
+
 @FunctionalInterface
 private interface OperationStrategy {
 	void apply(Operation operation, long handle, int zoom);
 }
 
 private abstract static class Operation {
-	private OperationStrategy operationStrategy;
+	private final OperationStrategy operationStrategy;
 
 	Operation(OperationStrategy operationStrategy) {
 		this.operationStrategy = operationStrategy;
 	}
 
-	void apply(long handle, int zoom) {
-		operationStrategy.apply(this, handle, zoom);
+	void apply(RegionHandle regionHandle) {
+		operationStrategy.apply(this, regionHandle.handle, regionHandle.zoom);
```

Comment:
NIT: ... since you're using records:

```suggestion
		operationStrategy.apply(this, regionHandle.handle(), regionHandle.zoom());
```

There are other places in the code too.

Reply:
Adapted in a01f1a88baaf56c2c7c80094bd49c13e5193c655